# Table of Contents
I Join 60s, 70s and 80s databases with a Top 20 songs per decade database

II Clean the data, shaping it to our needs

III Analyze the music data and utilize charts/graphs to find The Perfect Setlist


In [13]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import plotly as pypl
import plotly.express as px  

`#Read in Top 20 database, change name to track, Artist to artist, Decade to decade, Rank to rank, drop all other rows.
#Read in csvs *60s, 70s and 80s*, bring together, remove all songs with '0' target value. These songs are considered unlistenable for our needs.
Bring all 4 CSVs together
dataframes: remove 'target', 'remove the 'uri' column, change ms_duration from milliseconds to min:sec`

***
First thing...for each of these 3 decade CSVs I would like to add a column named 'decade' and assign the appropriate variable to said file, i.e., 60, 70, 80. Once all is good, we will combine these first three CSVs into one usable dataframe.
***

<img src="Groovy.png" style="margin-left:auto; margin-right:auto"/>

***

<center>DECADE: THE 1960s<center>
    
***

In [18]:
flower_df = pd.read_csv('dataset-of-60s.csv')

In [19]:
#add decade column and call 60

In [20]:
flower2_df = flower_df.assign(decade='60')

In [21]:
flower2_df.head()

track            artist  \
0  Jealous Kind Of Fella     Garland Green   
1          Initials B.B.  Serge Gainsbourg   
2           Melody Twist       Lord Melody   
3          Mi Bomba Sonó        Celia Cruz   
4            Uravu Solla       P. Susheela   

                                    uri  danceability  energy  key  loudness  \
0  spotify:track:1dtKN6wwlolkM8XZy2y9C1         0.417   0.620    3    -7.727   
1  spotify:track:5hjsmSnUefdUqzsDogisiX         0.498   0.505    3   -12.475   
2  spotify:track:6uk8tI6pwxxdVTNlNOJeJh         0.657   0.649    5   -13.392   
3  spotify:track:7aNjMJ05FvUXACPWZ7yJmv         0.590   0.545    7   -12.058   
4  spotify:track:1rQ0clvgkzWr001POOPJWx         0.515   0.765   11    -3.515   

   mode  speechiness  acousticness  instrumentalness  liveness  valence  \
0     1       0.0403         0.490          0.000000    0.0779    0.845   
1     1       0.0337         0.018          0.107000    0.1760    0.797   
2     1       0.0380         0.846          0.000004    0.1190    0.908   
3     0       0.1040         0.706          0.024600    0.0610    0.967   
4     0       0.1240         0.857          0.000872    0.2130    0.906   

     tempo  duration_ms  time_signature  chorus_hit  sections  target decade  
0  185.655       173533               3    32.94975         9       1     60  
1  101.801       213613               4    48.82510        10       0     60  
2  115.940       223960               4    37.22663        12       0     60  
3  105.592       157907               4    24.75484         8       0     60  
4  114.617       245600               4    21.79874        14       0     60

<img src="70s.png" style="margin-left:auto; margin-right:auto"/>

***

<center>DECADE: THE 1970s<center>
    
***

In [11]:
disco_df = pd.read_csv('dataset-of-70s.csv')

In [12]:
disco2_df = disco_df.assign(decade='70')

In [8]:
disco2_df.head()

track                artist  \
0            Porque?       Reginaldo Rossi   
1          Free Jazz  Maria Teresa Luciani   
2  Reject of Society                 Crass   
3             Yeah !              The Jets   
4   Gotta Find A Way           The Moments   

                                    uri  danceability  energy  key  loudness  \
0  spotify:track:28KC2wl7jDv6Ms1Uc0OMpm         0.669   0.547    1    -9.873   
1  spotify:track:0zagn8PNnM7pM8pz1gXOAD         0.291   0.300    1   -14.054   
2  spotify:track:18eNASEiuAhO2ML5LBIoG4         0.355   0.968    7    -4.687   
3  spotify:track:1UlJzHc8E5EprQFJxjSyPL         0.598   0.891    6   -11.904   
4  spotify:track:1Jtlirju8tUXTtvyZJAN5v         0.440   0.376    5   -10.939   

   mode  speechiness  acousticness  instrumentalness  liveness  valence  \
0     0       0.0576        0.3530             0.000    0.0515    0.730   
1     0       0.0326        0.9050             0.492    0.1180    0.039   
2     1       0.1890        0.0504             0.000    0.1300    0.768   
3     1       0.0552        0.0183             0.836    0.5790    0.826   
4     1       0.0482        0.3480             0.000    0.0815    0.358   

     tempo  duration_ms  time_signature  chorus_hit  sections  target decade  
0  131.073       141067               4    28.79763         6       0     70  
1   75.619       228224               4    44.10303         9       0     70  
2  181.291        66800               3    25.23004         6       0     70  
3  144.383       155395               4    55.62066         7       0     70  
4  130.438       218333               4    34.90979        12       1     70

<img src="80s-png-back-to-the-80s.png" style="margin-left:auto; margin-right:auto"/>

***

<center>DECADE: THE 1980s<center>
    
***

In [9]:
mcfly_df = pd.read_csv('dataset-of-80s.csv')

In [10]:
mcfly2_df = mcfly_df.assign(decade='80')

In [11]:
mcfly2_df.head()

track            artist  \
0         Walking Blues  Big Joe Williams   
1  Suddenly Last Summer        The Motels   
2             Sanctuary        Béla Fleck   
3        The Wild Rover        The Pogues   
4  In The Driver's Seat    John Schneider   

                                    uri  danceability  energy  key  loudness  \
0  spotify:track:1ZjN5X8LmUB67pWPgimW3B         0.509   0.277    6   -14.323   
1  spotify:track:4fLIM0B1WwrLux9RdnMvze         0.716   0.753    2    -5.682   
2  spotify:track:3DwlNfiCQSdj0GOxYkR9Rq         0.360   0.542    5   -13.885   
3  spotify:track:6JyYNPLalPgGa7XnclF5FO         0.656   0.512    7   -11.872   
4  spotify:track:6jJi8OXF5qaFdysB6sjWIT         0.642   0.889    2    -5.620   

   mode  speechiness  acousticness  instrumentalness  liveness  valence  \
0     1       0.0495         0.827           0.00206    0.0756    0.640   
1     1       0.0286         0.162           0.03060    0.0831    0.561   
2     1       0.0339         0.368           0.16500    0.1160    0.803   
3     1       0.0290         0.585           0.00000    0.0720    0.880   
4     0       0.0494         0.375           0.00000    0.1800    0.764   

     tempo  duration_ms  time_signature  chorus_hit  sections  target decade  
0  101.157       161893               4    68.46530         7       0     80  
1  120.141       222000               4    57.71583        11       1     80  
2  116.831       444907               4    30.34574        17       0     80  
3   97.500       157893               3    50.97022         7       0     80  
4  163.351       162293               4    33.62053         7       1     80

`This is data cleaning; Also want to make reference to the Data Dictionary intermittently when explaining`

In [29]:
combined_df = pd.concat([flower2_df, disco2_df, mcfly2_df], ignore_index=True)

In [30]:
combined_df.sample(10)

track  \
17832                              Islands In The Stream   
1978                                          Kanne Raja   
5939                                           Hey, Girl   
18756               Thrash 'Till Death - 2017 - Remaster   
1205                                      When You Dance   
19300  Return To The Present - Somewhere In Time/Soun...   
6426                             I Think We're Alone Now   
14364                    Mercy - 2006 Remastered Version   
10515                                            Caprice   
22996                                          Flash (1)   

                                    artist  \
17832  Kenny Rogers Duet With Dolly Parton   
1978                           P. Susheela   
5939                         Freddie Scott   
18756                              Tankard   
1205                   Jay & The Americans   
19300                           John Barry   
6426           Tommy James & The Shondells   
14364                                 Wire   
10515                       Claude Bolling   
22996                      Keith Mansfield   

                                        uri  danceability  energy  key  \
17832  spotify:track:4dqEzdFAYkj1rjYLVDuTlN         0.595   0.473    8   
1978   spotify:track:79NLqLkIIhlDk1Ku7XIM0J         0.483   0.341    5   
5939   spotify:track:3vZyw7jyBpwgIEExgEOGJd         0.394   0.297    5   
18756  spotify:track:5cmKPo1nG00RFSnPp3f6Hv         0.293   0.990   11   
1205   spotify:track:2oHcWtwRY6IEiKHCPrN5Ji         0.518   0.768    7   
19300  spotify:track:1qvLi2D0zErCp8VZLqXt1S         0.150   0.172    0   
6426   spotify:track:3AXDLGI5TIYi72JcYFvNpv         0.783   0.420   11   
14364  spotify:track:4jyZHdfC75dIxqJgovAYKv         0.458   0.688    4   
10515  spotify:track:1is7VpzZDPJo57z8R5j2Pt         0.424   0.242    0   
22996  spotify:track:1RzI5iGERF8oVNicZxXP65         0.633   0.349    5   

       loudness  mode  speechiness  acousticness  instrumentalness  liveness  \
17832    -9.828     1       0.0347      0.702000          0.000000    0.0725   
1978     -5.932     1       0.0319      0.963000          0.036800    0.1120   
5939    -12.867     1       0.0261      0.745000          0.000019    0.1760   
18756    -6.652     0       0.1220      0.000107          0.000000    0.3480   
1205     -7.373     1       0.0493      0.586000          0.000045    0.0811   
19300   -19.887     1       0.0426      0.641000          0.828000    0.1060   
6426    -11.050     0       0.1030      0.072200          0.031600    0.0692   
14364    -7.809     1       0.0311      0.024300          0.121000    0.0995   
10515   -14.458     1       0.0488      0.981000          0.177000    0.1010   
22996   -12.037     1       0.0413      0.015400          0.980000    0.1340   

       valence    tempo  duration_ms  time_signature  chorus_hit  sections  \
17832   0.7160  204.016       249293               4    16.88278        11   
1978    0.4700  116.063       202502               4    32.58161        11   
5939    0.5000   85.804       191227               4    25.76745        10   
18756   0.2790  174.676       152467               4    24.55115        10   
1205    0.7610  133.927       174040               4    26.42985        11   
19300   0.0389   80.211       251200               4    35.44226        12   
6426    0.7160  131.750       130320               4    38.31500         7   
14364   0.4360  116.494       346400               4    75.26332        13   
10515   0.2230  130.774       404000               4    55.75794        20   
22996   0.6480  154.241        61259               3    19.95541         4   

       target decade  
17832       1     80  
1978        0     60  
5939        1     60  
18756       0     80  
1205        1     60  
19300       0     80  
6426        0     60  
14364       0     70  
10515       0     70  
22996       0     80

***
**BITCHIN!** 

First, let's see if there are any missing variables.
***

In [31]:
combined_df.isna().sum()

track               0
artist              0
uri                 0
danceability        0
energy              0
key                 0
loudness            0
mode                0
speechiness         0
acousticness        0
instrumentalness    0
liveness            0
valence             0
tempo               0
duration_ms         0
time_signature      0
chorus_hit          0
sections            0
target              0
decade              0
dtype: int64

**Alright...Alright!**

No missing variables. Next, let check our types of data and the size.


In [32]:
combined_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23316 entries, 0 to 23315
Data columns (total 20 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   track             23316 non-null  object 
 1   artist            23316 non-null  object 
 2   uri               23316 non-null  object 
 3   danceability      23316 non-null  float64
 4   energy            23316 non-null  float64
 5   key               23316 non-null  int64  
 6   loudness          23316 non-null  float64
 7   mode              23316 non-null  int64  
 8   speechiness       23316 non-null  float64
 9   acousticness      23316 non-null  float64
 10  instrumentalness  23316 non-null  float64
 11  liveness          23316 non-null  float64
 12  valence           23316 non-null  float64
 13  tempo             23316 non-null  float64
 14  duration_ms       23316 non-null  int64  
 15  time_signature    23316 non-null  int64  
 16  chorus_hit        23316 non-null  float6

In [33]:
combined_df.shape

(23316, 20)

# **That's Heavy!**

Now that we can see the number of columns (20), their datatypes (dtype) and the number of rows (23,316), let's go ahead and create an index and then we can start WEEDing out unnecessary data.  

In [34]:
combined_df.columns

Index(['track', 'artist', 'uri', 'danceability', 'energy', 'key', 'loudness',
       'mode', 'speechiness', 'acousticness', 'instrumentalness', 'liveness',
       'valence', 'tempo', 'duration_ms', 'time_signature', 'chorus_hit',
       'sections', 'target', 'decade'],
      dtype='object')

**RAD...**

So, the first thing we will do is get rid of the 'uri' column in the above index. It's a link that takes you directly to the Spotify application without having to go through the web page first...AND unnecessary for what we're after. Second, we will remove all rows with '0s' under 'target' as these were deemed 'unlistenable songs'(see data dictionary for explanation) so, also, not what we're after. Finally, we will eventually rename 'duration_ms' and call it 'min:sec'. We are going to change those wicked milliseconds into 'real time' so we don't have to 'math' cuz 'math' is hard.

In [35]:
combined_df.drop(columns=combined_df.columns[2], inplace=True)
combined_df
combined_df.head()


track            artist  danceability  energy  key  \
0  Jealous Kind Of Fella     Garland Green         0.417   0.620    3   
1          Initials B.B.  Serge Gainsbourg         0.498   0.505    3   
2           Melody Twist       Lord Melody         0.657   0.649    5   
3          Mi Bomba Sonó        Celia Cruz         0.590   0.545    7   
4            Uravu Solla       P. Susheela         0.515   0.765   11   

   loudness  mode  speechiness  acousticness  instrumentalness  liveness  \
0    -7.727     1       0.0403         0.490          0.000000    0.0779   
1   -12.475     1       0.0337         0.018          0.107000    0.1760   
2   -13.392     1       0.0380         0.846          0.000004    0.1190   
3   -12.058     0       0.1040         0.706          0.024600    0.0610   
4    -3.515     0       0.1240         0.857          0.000872    0.2130   

   valence    tempo  duration_ms  time_signature  chorus_hit  sections  \
0    0.845  185.655       173533               3    32.94975         9   
1    0.797  101.801       213613               4    48.82510        10   
2    0.908  115.940       223960               4    37.22663        12   
3    0.967  105.592       157907               4    24.75484         8   
4    0.906  114.617       245600               4    21.79874        14   

   target decade  
0       1     60  
1       0     60  
2       0     60  
3       0     60  
4       0     60

In [36]:
new_df = combined_df[combined_df['target'] !=0]
new_df.head()

track          artist  danceability  energy  key  \
0   Jealous Kind Of Fella   Garland Green         0.417   0.620    3   
7              Happy Days    Marv Johnson         0.720   0.624    5   
9        I Can Hear Music  The Beach Boys         0.511   0.603    2   
11              Ride Away     Roy Orbison         0.520   0.411    9   
13               P.T. 109      Jimmy Dean         0.746   0.666    2   

    loudness  mode  speechiness  acousticness  instrumentalness  liveness  \
0     -7.727     1       0.0403        0.4900          0.000000    0.0779   
7     -9.086     0       0.0473        0.7950          0.000000    0.4880   
9     -7.637     1       0.0280        0.0385          0.000002    0.1420   
11   -12.411     1       0.0315        0.7860          0.001200    0.1460   
13   -10.408     1       0.1460        0.5430          0.000000    0.0488   

    valence    tempo  duration_ms  time_signature  chorus_hit  sections  \
0     0.845  185.655       173533               3    32.94975         9   
7     0.887  119.999       160040               4    30.42891         8   
9     0.685  128.336       157293               4    43.36534         9   
11    0.350  120.290       207573               4    37.54527         9   
13    0.840   94.024       192427               4    78.59848         5   

    target decade  
0        1     60  
7        1     60  
9        1     60  
11       1     60  
13       1     60

`new_df.to_csv('Three Decades.csv', index=False, encoding='utf-8')
Mark out csv after writing to be safe`

*** 
**GNARLY!**
Now everything in this dataframe is at least considered 'listenable' and we're done. I wrote the combined 'decade' file to a new CSV. Next, we're going to read in the 'Top 20 All Decades' file. Here we will remove everything but the columns: Name, Artist, Decade and Rank. We will also go ahead and change their names to read the exact same way as the other combined file to make life easier and ultimately write this to a CSV.
***

In [37]:
top_df = pd.read_csv('Top 20 All Decades.csv')
top_df
top_df.head()

Name                       Artist  Decade  Rank  \
0                    The Twist               Chubby Checker      60     1   
1                     Hey Jude                  The Beatles      60     2   
2  Theme from “A Summer Place”  Percy Faith & His Orchestra      60     3   
3          Tossin’ and Turnin’                  Bobby Lewis      60     4   
4     I Want to Hold Your Hand                  The Beatles      60     5   

   danceability  energy  key  loudness  speechiness  acousticness  \
0         0.533   0.638    4    -7.130       0.0341         0.202   
1         0.462   0.632    5    -7.438       0.0248         0.091   
2         0.466   0.389    5   -12.825       0.0253         0.631   
3         0.569   0.722    0    -8.914       0.0393         0.255   
4         0.490   0.715    7    -5.549       0.0476         0.386   

   instrumentalness  liveness  valence    tempo  duration_ms  time_signature  \
0             0.000    0.0729    0.937  156.663       153760               4   
1             0.000    0.3970    0.585   74.268       238854               4   
2             0.843    0.2950    0.749   92.631       144893               4   
3             0.000    0.1080    0.965  142.736       159640               4   
4             0.000    0.3110    0.866  130.726       145747               4   

         Country          City  
0  United States  Philadelphia  
1        England     Liverpool  
2         Canada       Toronto  
3  United States  Indianapolis  
4        England     Liverpool

In [38]:
top_df = top_df.rename(columns={"Name": "track", "Artist": "artist", "Decade": "decade", "Rank": "rank"})

In [39]:
top_df.head()

track                       artist  decade  rank  \
0                    The Twist               Chubby Checker      60     1   
1                     Hey Jude                  The Beatles      60     2   
2  Theme from “A Summer Place”  Percy Faith & His Orchestra      60     3   
3          Tossin’ and Turnin’                  Bobby Lewis      60     4   
4     I Want to Hold Your Hand                  The Beatles      60     5   

   danceability  energy  key  loudness  speechiness  acousticness  \
0         0.533   0.638    4    -7.130       0.0341         0.202   
1         0.462   0.632    5    -7.438       0.0248         0.091   
2         0.466   0.389    5   -12.825       0.0253         0.631   
3         0.569   0.722    0    -8.914       0.0393         0.255   
4         0.490   0.715    7    -5.549       0.0476         0.386   

   instrumentalness  liveness  valence    tempo  duration_ms  time_signature  \
0             0.000    0.0729    0.937  156.663       153760               4   
1             0.000    0.3970    0.585   74.268       238854               4   
2             0.843    0.2950    0.749   92.631       144893               4   
3             0.000    0.1080    0.965  142.736       159640               4   
4             0.000    0.3110    0.866  130.726       145747               4   

         Country          City  
0  United States  Philadelphia  
1        England     Liverpool  
2         Canada       Toronto  
3  United States  Indianapolis  
4        England     Liverpool

In [40]:
top_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99 entries, 0 to 98
Data columns (total 18 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   track             99 non-null     object 
 1   artist            99 non-null     object 
 2   decade            99 non-null     int64  
 3   rank              99 non-null     int64  
 4   danceability      99 non-null     float64
 5   energy            99 non-null     float64
 6   key               99 non-null     int64  
 7   loudness          99 non-null     float64
 8   speechiness       99 non-null     float64
 9   acousticness      99 non-null     float64
 10  instrumentalness  99 non-null     float64
 11  liveness          99 non-null     float64
 12  valence           99 non-null     float64
 13  tempo             99 non-null     float64
 14  duration_ms       99 non-null     int64  
 15  time_signature    99 non-null     int64  
 16  Country           99 non-null     object 
 17 

In [41]:
top_df.head()

track                       artist  decade  rank  \
0                    The Twist               Chubby Checker      60     1   
1                     Hey Jude                  The Beatles      60     2   
2  Theme from “A Summer Place”  Percy Faith & His Orchestra      60     3   
3          Tossin’ and Turnin’                  Bobby Lewis      60     4   
4     I Want to Hold Your Hand                  The Beatles      60     5   

   danceability  energy  key  loudness  speechiness  acousticness  \
0         0.533   0.638    4    -7.130       0.0341         0.202   
1         0.462   0.632    5    -7.438       0.0248         0.091   
2         0.466   0.389    5   -12.825       0.0253         0.631   
3         0.569   0.722    0    -8.914       0.0393         0.255   
4         0.490   0.715    7    -5.549       0.0476         0.386   

   instrumentalness  liveness  valence    tempo  duration_ms  time_signature  \
0             0.000    0.0729    0.937  156.663       153760               4   
1             0.000    0.3970    0.585   74.268       238854               4   
2             0.843    0.2950    0.749   92.631       144893               4   
3             0.000    0.1080    0.965  142.736       159640               4   
4             0.000    0.3110    0.866  130.726       145747               4   

         Country          City  
0  United States  Philadelphia  
1        England     Liverpool  
2         Canada       Toronto  
3  United States  Indianapolis  
4        England     Liverpool

In [42]:
top_df.shape

(99, 18)

In [43]:
top_df2 = top_df.drop(top_df.columns[4:18], axis=1)

In [44]:
top_df2.head()

track                       artist  decade  rank
0                    The Twist               Chubby Checker      60     1
1                     Hey Jude                  The Beatles      60     2
2  Theme from “A Summer Place”  Percy Faith & His Orchestra      60     3
3          Tossin’ and Turnin’                  Bobby Lewis      60     4
4     I Want to Hold Your Hand                  The Beatles      60     5

In [45]:
top_df3 = top_df2[~((top_df2['decade'] == 90) | (top_df2['decade'] == 2000))]

In [46]:
top_df3.sample(40)

track  \
15                                 Sugar Shack   
57                             Say You, Say Me   
40                            Bette Davis Eyes   
55                                 Abracadabra   
27  Tie a Yellow Ribbon Round the Ole Oak Tree   
43                       Every Breath You Take   
3                          Tossin’ and Turnin’   
33                                  My Sharona   
17            (Sittin’ On) The Dock of the Bay   
21                                     Le Feak   
0                                    The Twist   
11                                   I’m Sorry   
34             Killing Me Softly With His Song   
47                                     Call Me   
50                   (Just Like) Starting Over   
23           I Just Want to Be Your Everything   
20      Tonight’s The Night (Gonna Be Alright)   
39                                    Physical   
41                                Endless Love   
35                             Best of My Love   
19                        You Light Up My Life   
53                 That’s What Friends Are For   
45                  Another One Bites the Dust   
52                               Ebony & Ivory   
16                           Honky Tonk Woman    
36                             The Way We Were   
37                        A Fifth of Beethoven   
22                       How Deep is Your Love   
18                                Big Bad John   
5                               I’m a Believer   
8             I Heard it Through the Grapevine   
2                  Theme from “A Summer Place”   
13                                Hello Dolly!   
24                            Silly Love Songs   
46                                 Say Say Say   
7                                 Sugar, Sugar   
6                Aquarius/Let the Sun Shine In   
42                            Eye Of The Tiger   
28                              Shadow Dancing   
31                              You’re So Vain   

                              artist  decade  rank  
15    Jimmy Gilmer and the Fireballs      60    17  
57                     Lionel Richie      80    19  
40                        Kim Carnes      80     2  
55                 Steve Miller Band      80    17  
27             Dawn ft. Tony Orlando      70     9  
43                        The Police      80     5  
3                        Bobby Lewis      60     4  
33                         The Knack      70    15  
17                      Otis Redding      60    19  
21                              Chic      70     3  
0                     Chubby Checker      60     1  
11                        Brenda Lee      60    13  
34                     Roberta Flack      70    16  
47                           Blondie      80     9  
50                       John Lennon      80    12  
23                         Andy Gibb      70     5  
20                       Rod Stewart      70     2  
39                Olivia Newton-John      80     1  
41        Lionel Richie & Diana Ross      80     3  
35                      The Emotions      70    17  
19                      Debbie Boone      70     1  
53                    Dionne Warwick      80    15  
45                             Queen      80     7  
52    Paul McCartney & Stevie Wonder      80    14  
16                The Rolling Stones      60    18  
36                  Barbra Streisand      70    18  
37                     Walter Murphy      70    19  
22                          Bee Gees      70     4  
18                        Jimmy Dean      60    20  
5                        The Monkees      60     6  
8                        Marvin Gaye      60     9  
2        Percy Faith & His Orchestra      60     3  
13                   Louis Armstrong      60    15  
24                             Wings      70     6  
46  Paul McCartney & Michael Jackson      80     8  
7                        The Archies      60     8  
6                The Fifth Di

***
Now, I will go ahead and write the second file and head for the second Notebook entitled, 'Perfect Setlist Part II'
***
`#top_df3.to_csv('Top Twenty.csv', index=False, encoding='utf-8')
#Mark out csv after writing...just to be safe from overwriting...jumping back and forth. Fool me twenty times, shame on me` 